### DISCUSSION TITLE
Start a node programatically from from another Go application.

### DISCUSSION THREAD

**Discussion by bradub** [btcsuite/btcd#2052] [Q&A]
Hello, do you think it's feasible to start a btcd server from another Go program by importing the necessary dependencies for starting the server? Or is there too much unexported functionality in order to achieve this?

**Comment by guggero** [comment]
I'm not extremely familiar with the code base, so not sure if there are many un-exported things you'd need for that.
But I think it should be pretty easy to figure out. Just copy the code of `btcdMain()` (https://github.com/btcsuite/btcd/blob/master/btcd.go#L44C9-L44C9) into your project and see what's missing (probably the whole config parsing functionality, but you'd likely want to supply a fully populated config from your code anyway and not inject a file based config).
